### PR TITLE
[dist] Add OBS Service Dispatch Service for Systemd

### PR DESCRIPTION
**Pull request by leemgs**
Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>

### CODE REVIEW

**Comment by leemgs** [comment]
* After this PR:

```
root@mytizen:~# systemctl status obsservicedispatch.service 
● obsservicedispatch.service - OBS Service Dispatch
   Loaded: loaded (/lib/systemd/system/obsservicedispatch.service; enabled; vendor preset: enabled)
   Active: active (exited) since 화 2017-04-18 19:12:32 KST; 11min ago
  Process: 5357 ExecStart=/bin/bash -c /bin/echo -n 'Initializing obsservicedispatch' (code=exited, status=0/SUCCESS)
  Process: 5353 ExecStartPre=/bin/mkdir -p $logdir (code=exited, status=0/SUCCESS)
 Main PID: 5357 (code=exited, status=0/SUCCESS)
    Tasks: 0
   Memory: 0B
      CPU: 0
   CGroup: /system.slice/obsservicedispatch.service

 4월 18 19:12:32 mytizen systemd[1]: Starting OBS Service Dispatch...
 4월 18 19:12:32 mytizen bash[5357]: Initializing obsservicedispatch
 4월 18 19:12:32 mytizen systemd[1]: Started OBS Service Dispatch.
root@mytizen:~# 

```

**Comment by leemgs** [comment]
This PR is to fix https://github.com/openSUSE/open-build-service/issues/2997.

File: dist/systemd/obsservicedispatch.service [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+[Service]
+EnvironmentFile=/etc/sysconfig/obs-server
+Environment=logdir=/srv/obs/log
+Type=oneshot
+ExecStart=/bin/bash -c "/bin/echo -n 'Initializing obsservicedispatch'"
+ExecStartPre=/bin/mkdir -p $logdir
+ExecStart=bsuser=`/usr/bin/perl -I/usr/lib/obs/server -MBSConfig -e 'print ( $BSConfig::bsuser || "obsrun" )'`
+ExecStart=bsgroup=`/usr/bin/perl -I/usr/lib/obs/server -MBSConfig -e 'print ( $BSConfig::bsgroup || "obsrun" )'`
+ExecStart=chown $bsuser:$bsgroup "$logdir"
```
Comment (k0da) [inline dist/systemd/obsservicedispatch.service]: IMO, this must be static configuration and shouldn't be done on each service start.

Comment (leemgs) [inline dist/systemd/obsservicedispatch.service]: It is a systemd-based service file  that is created based on sysvinit-based  obsservicedispatch file (e.g., ./dist/obsservicedispatch).

**Comment by M0ses** [comment]
Thanks for your engagement and contribution.

As already stated in some other PR`s, we will need some time to evaluate a consistent plan to merge from systemV to systemd, as the startup mechanisms are very critical . This commit looks very promising to me but it needs further testing

I will setup an test environment to test this change for

* new installation via appliance
* new installation via rpm
* various ways of updates

BTW:
Instead of using the "perl -I ...." commands, you could also use e.g.

obs_admin --query-config logdir

as it should be used as cli-interface to obs settings in future.

**Comment by leemgs** [comment]
>obs_admin --query-config logdir

Thank you for sharing "obs_admin" command. :) BTW, I coul not find "--query-config" option from http://openbuildservice.org/files/manuals/obs-admin-guide.pdf and "$ obs_admin | grep query-config". 
Alternatively, we can also get the data from /etc/obs-server file as following: 
```
EnvironmentFile=/etc/obs-server
ExecStartPre=/bin/mkdir -p $OBS_LOG_DIR
```
 * Reference (/etc/obs-server) : http://openbuildservice.org/help/manuals/obs-admin-guide/obs.cha.overview_filesystem.html#_etc_sysconfig_obs_server

**Comment by M0ses** [comment]
Sorry - I cannot reproduce your "obs_admin missing help for --query-config" issue:
```
obs-server:/tmp/worker # obs_admin | grep query-config
 --query-config <variable>
```

but I have to admit, that the admin guides printout of "obs_admin --help" is quite useless if it is outdated - maybe we should remove the output there

**Comment by M0ses** [comment]
"obs_admin --query-config" will always win over /etc/sysconfig/obs-server because it print the internally used variables. If nothing is specified in /etc/sysconfig/obs-server, the internal default values are used - and this values can only be queried by "obs_admin --query-config <variable>"

**Pull request closed without merging**